src/RequestQueue.js

```diff
@@ -1,34 +1,83 @@
-'use strict'; 
+'use strict';
 
 import SuperAgent from 'superagent';
 
 function RequestQueue( superAgent ) {
-	
+
 	let Request = superAgent.Request;
 	let superAgentEnd = Request.prototype.end;
-
 	let queue = [];
 
-	//Reset request to allow retry
-	function reset( obj, timeout ) {
-		
-		delete obj.xhr;
-		delete obj._timer;
-		delete obj.aborted;
-		delete obj.timedout;
-		delete obj.timeout;
+	const TIMEOUT_REGEX = /timeout of \d+ms exceeded/;

```

Comment:
Superagent returns these constructed errors so it is only possible to test the string that is returned as there is no object being returned. For timeouts the only time I have been able to retrieve an err code is during a timeout is when testing with a local expressjs server.